File: frontend/src/app/auth/auth.models.ts
export type UserRole = 'Admin' | 'Planner' | 'Viewer';

export interface AuthUser {
  userId: string;
  email: string;
  firstName: string;
  lastName: string;
  role: UserRole;
}

export interface AuthResponse {
  token: string;
  user: AuthUser;
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface SignupRequest extends LoginRequest {
  firstName: string;
  lastName: string;
}

File: api/Models/Auth/AuthModels.cs
namespace Naologic_API.Models.Auth;

public sealed record SignupRequest(string Email, string Password, string FirstName, string LastName);

public sealed record LoginRequest(string Email, string Password);

public sealed record AuthUserResponse(Guid UserId, string Email, string FirstName, string LastName, string Role);

public sealed record AuthResponse(string Token, AuthUserResponse User);

public sealed record AppUserAccount(
    Guid UserId,
    string Email,
    string PasswordHash,
    string FirstName,
    string LastName,
    string Role,
    bool IsActive,
    DateTime CreatedAt);
